src/plugins/intel_cpu/src/emitters/snippets/aarch64/cpu_generator.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/snippets_isa.hpp"
+#include "cpu_generator.hpp"
+#include "jit_snippets_emitters.hpp"
+#include "emitters/plugin/aarch64/jit_eltwise_emitters.hpp"
+#include "emitters/snippets/aarch64/jit_kernel_emitter.hpp"
+#include "emitters/snippets/aarch64/jit_loop_emitters.hpp"
+#include "emitters/snippets/aarch64/jit_memory_emitters.hpp"
+#include "emitters/snippets/aarch64/jit_fill_emitter.hpp"
+
+#include "openvino/opsets/opset13.hpp"
+
+namespace ov {
+
+#define CREATE_SNIPPETS_EMITTER(e_type) { \
+    [this](const snippets::lowered::ExpressionPtr& expr) -> std::shared_ptr<snippets::Emitter> { \
+        return std::make_shared<e_type>(h.get(), isa, expr); \
+    }, \
+    [](const std::shared_ptr<ov::Node>& n) -> std::set<std::vector<element::Type>> { \
+        return e_type::get_supported_precisions(n); \
+    } \
+}
+
+#define CREATE_CPU_EMITTER(e_type) { \
+    [this](const snippets::lowered::ExpressionPtr& expr) -> std::shared_ptr<snippets::Emitter> { \
+        return std::make_shared<e_type>(h.get(), isa, expr->get_node()); \
+    }, \
+    [](const std::shared_ptr<ov::Node>& n) -> std::set<std::vector<element::Type>> { \
+        return e_type::get_supported_precisions(n); \
+    } \
+}
+
+class jit_snippet : public dnnl::impl::cpu::aarch64::jit_generator {
+public:
+    DECLARE_CPU_JIT_AUX_FUNCTIONS(jit_snippet)
+
+    ~jit_snippet() = default;
```

Comment:
Thanks Edward for the comment! I think the main reason for class jit_snippets to inherit dnnl::impl::cpu::aarch64::jit_generator, is to override generate() method to be empty, so that jit code will be only generated by emitters directly for Snippets. So this jit_snippets class should not be inherited. If this is not what you are concerning, please elaborate a little bit. Thanks!

Reply:
While `jit_snippet` is inherited, the destructor has to be virtual. You can have not memory leaks right now but it will be more secure. 
Thanks.

Reply:
Applied. Thanks Edward!